snapcraft/internal/elf.py

```diff
@@ -504,8 +504,10 @@ def _get_rpath(self, elf_file) -> str:
                     rel_library_path_dir = os.path.dirname(rel_library_path)
                     # return the dirname, with the first .. replace
                     # with $ORIGIN
-                    origin_rpaths.append(rel_library_path_dir.replace(
-                        '..', '$ORIGIN', 1))
+                    origin_rpath = rel_library_path_dir.replace(
+                        '..', '$ORIGIN', 1)
+                    if origin_rpath not in origin_rpaths:
+                        origin_rpaths.append(origin_rpath)
```

Comment:
Is there a reason this isn't just a set? Then we don't need to check it at all.

Reply:
order is important and the set would disorganize that

Reply:
Oh duh, that would be bad indeed.